### PR TITLE
oop-programs_with_interesting_classes-easy-q8

### PR DESCRIPTION
**Pull request by KaiKhademi**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”

### CODE REVIEW

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]
Please see my feedback on other questions in this category and apply it to this PR.

File: src/oop/programs_with_interesting_classes/easy/q8/QuestionText.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# oop-programs_with_interesting_classes-easy-q8
+
+- Parsons Question
+
+## Question Text
+
+You are environmentally conscious, and you want to know how much water your daily activities consume. Make a class
+called `Activity` with the attributes **name**, **consumptionRate**, and **duration**
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q8/QuestionText.md:8]:
```suggestion
called `Activity` to track your water usage using the requirements below. Then make a `TestWaterConsumption` class to test `Activity`.
```

File: src/oop/programs_with_interesting_classes/easy/q8/QuestionText.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    - A `String` attribute called **name**
+    - A `double` attribute called **consumptionRate**
+    - An `int` attribute called **duration**
+
+- Constructors:
+    - A constructor visible from another class that accepts a `String`, a `double`, and an `int` as parameters, and uses
+      the values to initialize the **name**, **consumptionRate**, and **duration** attributes
+    - Note that this constructor should be visible from all other classes.
+
+- Methods:
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q8/QuestionText.md:25]: make note of visibility

File: src/oop/programs_with_interesting_classes/easy/q8/QuestionText.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    - An accessor method for the **name** attribute called `getName`.
+    - A `usageCalculator` method which return the double value of **consumptionRate** times **duration**.
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q8/QuestionText.md:27]:
```suggestion
    - A `usageCalculator` method which returns a double. The returned value should be **consumptionRate** multiplied by **duration**.
```

File: src/oop/programs_with_interesting_classes/easy/q8/QuestionText.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+### TestWaterConsumption class
+
+- Main method:
+    - In the main method create 3 `Activity` classes and then print the sum of water consumption for all of them, then
+      find the activity with the worst (highest) water consumption and then print it.
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q8/QuestionText.md:33]:
```suggestion
    - In the main method create 3 `Activity` classes and then print the sum of water consumption for all of them.
      Find the activity with the worst (highest) water consumption and then print it.
```

File: src/oop/programs_with_interesting_classes/easy/q8/MainTest.java [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+        assertTrue(testClass.hasReturnType(getAttributeName1, null, String.class), String.format(incorrectReturnType, getAttributeName1, className));
+        Object[][] arguments = {
+                {value1, String.class},
+                {value2, double.class},
+                {value3, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        String[] getMethodModifiers = {"public"};
+        String incorrectGetMethodMessage = String.join(" ",
+                "Your", getAttributeName1, "does not correctly get the value of the", attributeName1, "attribute.");
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q8/MainTest.java:137]:
```suggestion
                "Your", getAttributeName1, "method does not correctly get the value of the", attributeName1, "attribute.");
```

File: src/oop/programs_with_interesting_classes/easy/q8/MainTest.java [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+        Object getMethodOutput = testClass.callMethod(getAttributeName1, getMethodModifiers, classInstance);
+        _assertEquals(value1, getMethodOutput, incorrectGetMethodMessage);
+    }
+
+    @Test
+    @Tag("dependent1")
+    public void correctMainMethod() throws Throwable {
+        Clause[] clauses = {
+                new StringLiteral("The total water usage is 315.0"),
```
Comment (SamamthaH) [inline src/oop/programs_with_interesting_classes/easy/q8/MainTest.java:146]: make these strings variables at the top of the tests

**Review by SamamthaH — CHANGES_REQUESTED** [review]

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]
See Sam's review

**Pull request merged**